### PR TITLE
[IMP] sales: updated entire invoicing based on milestones doc

### PR DESCRIPTION
**Pull request by ksc-odoo**
PROJECT TASK: https://www.odoo.com/web#id=3475673&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form

### FWP up to `master`

### CODE REVIEW

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (jero-odoo) [inline content/applications/sales/sales/invoicing/milestone.rst:79]: I would specify that the Workspace template is for the Documents app and not the Projects app.

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (jero-odoo) [inline content/applications/sales/sales/invoicing/milestone.rst:116]:
```suggestion
When the order is confirmed, new
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (jero-odoo) [inline content/applications/sales/sales/invoicing/milestone.rst:125]: When I click on the smart button, the milestone page is not blank, it populated a list milestones from the SO, which I think is how it should work, right? It looks like you can then add others even from different SOs but I don't think the page should be blank?

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (jero-odoo) [inline content/applications/sales/sales/invoicing/milestone.rst:149]: Same as above, this should happen automatically, right?

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (jero-odoo) [inline content/applications/sales/sales/invoicing/milestone.rst:192]:
```suggestion
default :guilabel:`Regular Invoice` selection, and click the :guilabel:`Create Draft Invoice` button.
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (jero-odoo) [inline content/applications/sales/sales/invoicing/milestone.rst:221]:
```suggestion
Then, return to the sales order, via the breadcrumb links. Back on the sales order, in the
```

**Review by jero-odoo — CHANGES_REQUESTED** [review]
@ksc-odoo Great work! Just a few notes, let me know if you have any questions. 👍

**Comment by ksc-odoo** [comment]
Thanks for all the awesome feedback @jero-odoo -- super helpful, as always 👍 I made all of those suggested edits, so whenever you get a chance, if you could give this a quick "once-over" -- and let me know if you approve -- that would greatly appreciated. Once I hear back from you about that, I'll go ahead and tag the Materials team for a second round of review. Thanks!

**Comment by jero-odoo** [comment]
@ksc-odoo looks great to me 👍

**Review by jero-odoo — APPROVED** [review]

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (Felicious) [inline content/applications/sales/sales/invoicing/milestone.rst:99]: while creating the product to test your flow, I wasn't too sure which boxes to tick. I'm not sure if it's necessary to specify this information here-- I just wanted to point out everything I observed while following your doc (:
image

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (Felicious) [reply content/applications/sales/sales/invoicing/milestone.rst:125]: I might have missed a step, since I created these milestone products on an empty/new Project, but after clicking the "Milestone" smart button, the page *was* blank, and not pre-populated with milestones. If there's additional setup for this, maybe it would be helpful for us to include. Do we have another doc that explains how to create milestones for a project, so it'll be populated at this time? It would be useful to link it here (:

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (Felicious) [inline content/applications/sales/sales/invoicing/milestone.rst:135]: Again, like my previous comment, but I might have missed a step so my flow is all wonky now, but I manually created a new milestone using the New button, and i was able to edit the Quantity field (16.3 runbot)
image

Edited it again:
image

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (Felicious) [inline content/applications/sales/sales/invoicing/milestone.rst:163]: Totally optional formatting suggestion! It's just that Zac once replaced my double hypen (--) with an em dash (—), so I'll just do that here (:
```suggestion
To do that, first return to the sales order — either via the breadcrumb links, or by navigating to
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (Felicious) [inline content/applications/sales/sales/invoicing/milestone.rst:173]:
```suggestion
Next, return to the sales order — either by clicking :guilabel:`View Sales Order` on the
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (Felicious) [inline content/applications/sales/sales/invoicing/milestone.rst:178]: Not sure if this suggestion is a bit more vague than your previous sentence, but I was slightly confused by the wording "Delivered filled". Hope this rewording is helpful to you! Feel free to cherry-pick your favorite parts from your original and the reworded versions!

```suggestion
Back on the sales order, the milestone line item is marked as :guilabel:`Delivered` filled. That's
because the milestone has been reached.
```

**Review by Felicious — APPROVED** [review]
Hi @ksc-odoo! This is a really in-depth and well-written doc about the end-to-end process of the milestone workflow. I particularly appreciate that you included a note about the setup instructions of products before starting the flow. *Really* good idea and well-executed method of guiding readers through your doc :blush: 

After considering my optional suggestions to *further* refine your doc, it's definitely ready for final review :smile:

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (ksc-odoo) [reply content/applications/sales/sales/invoicing/milestone.rst:99]: As always, I sincerely appreciate your attention to detail. If memory serves, I don't believe it's required to tick any of these boxes, specifically. So, I think for the sake of simplicity, I'll just omit such a sentence to avoid any confusion. But, once again, thanks for pointing this out. Also, if you *do* think it's important to mention this, just let me know, and I'll add it in right away. Thanks! 👍

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (ksc-odoo) [reply content/applications/sales/sales/invoicing/milestone.rst:125]: I think you're right @Felicious 👍 When I was writing this doc, and going through the flow, I was working off an empty/new Project, and the page was, indeed, blank when the Milestone button was hit. So, perhaps JERO used a populated runbot during that review. So, that said, I think I'll revert it back to "blank" instead of "populated" -- so the reader following along, and *also* working off a new Project, won't be confused when they see a blank page.  Thanks for pointing this out! 👍

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (ksc-odoo) [reply content/applications/sales/sales/invoicing/milestone.rst:135]: Thanks for pointing this out! For the sake of avoiding any confusion, I'll go ahead and remove this minor note admonition. I really appreciate the in-depth feedback, @Felicious

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (ksc-odoo) [reply content/applications/sales/sales/invoicing/milestone.rst:178]: Great catch. I forgot to add the word "column" after "Delivered." I made that adjustment, and it reads much smoother now. Thanks again for pointing that out. 👍

**Comment by ksc-odoo** [comment]
Thanks for the always-thoughtful feedback, @Felicious -- I really appreciate it. I made all those suggested edits, and believe it's now ready for Final Review @odoo/us-doc-review 👍

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/milestone.rst:48]: Remove extra spaces in note
```suggestion
.. note::
   Leaving it on the default :guilabel:`Nothing` option won't negatively affect the desired
   workflow. However, a project *must* then be created directly from a sales order form with that
   specific product. Once a project is created *then* milestones and tasks can be created and
   configured.
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/milestone.rst:55]: Escape ampersand
```suggestion
- :guilabel:`Project \& Task`: Odoo creates a project and task related to this milestone product in
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/milestone.rst:67]: Escape ampersand
```suggestion
When :guilabel:`Project \& Task` or :guilabel:`Project` is selected, two new fields appear:
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/milestone.rst:105]: Alignment in note admonition 
```suggestion
   .. image:: milestone/settings-for-workflow.png
      :align: center
      :alt: Product with Service "Product Type" and "Task" in the Create on Order field on form.

   Those tasks are then attached to a pre-existing :guilabel:`Project`, which, in this case, is
   titled, :guilabel:`Rebranding Projects`.
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/milestone.rst:121]: Ideally this image should be cropped, there is no need to show the whole UI here and the smartbuttons are illegible. Could even remove this image :)
```suggestion
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/milestone.rst:242]: Update doc paths to be relative
```suggestion
   - :doc:`time_materials`
   - :doc:`proforma`
   - :doc:`invoicing_policy`
```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @ksc-odoo, this is a great PR! Nice job improving the content in this doc 👍 I have a few corrections and suggestions after reviewing, please see below. Tag me for another look once addressed, thank you!

**Comment by ksc-odoo** [comment]
Hey @samueljlieber ! Thanks for the awesome feedback! I made all those suggested edits, and believe it's ready for another look from you (whenever you get a chance) -- Thanks again 👍

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (StraubCreative) [inline content/applications/sales/sales/invoicing/milestone.rst:7]: Don't need the grammatical conjunction here.
The main subject of this section is projects so stick with that 😉 
```suggestion
completion of a project.
```

File: content/applications/sales/sales/invoicing/milestone.rst
Comment (StraubCreative) [inline content/applications/sales/sales/invoicing/milestone.rst:18]: Because the instruction is explicitly navigational, the conjunction we want here is "and" because it includes the creation AND configuration of products to work as milestones.

```suggestion
To create and configure products to work like this, first navigate to :menuselection:`Sales app
```

**Review by StraubCreative — APPROVED** [review]
Great doc @ksc-odoo. Very thorough 👍 
Approving with a couple comments below.

@robodoo r+

**Pull request closed without merging**
